### PR TITLE
Push code

### PR DESCRIPTION
**Pull request by tayyabshabab**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+import "antd/dist/antd.css";
+import "./App.scss";
+
+export class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: false,
+      mapLoading: true,
```
Comment (tayyabshabab) [inline src/App.js]: `mapLoading` can be removed

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+      locationsList: {},
+      formAction: "add",
```
Comment (tayyabshabab) [inline src/App.js]: Use `isNewLocation: true` instead

File: src/App.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        message.error("Something went wrong. We are looking into it.");
+      });
+  };
+
+  handleDelete = id => {
+    this.setState({ loading: true });
+    locationsApi
+      .deleteLocation(id)
+      .then(({ data }) => {
+        this.setState({ loading: false });
```
Comment (tayyabshabab) [inline src/App.js]: This can be removed

File: src/App.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+              </Button>
+              <Divider />
+              <LocationsListing
+                locations={locationsList.locations}
+                onEdit={this.showModal}
+                onDelete={this.handleDelete}
+              />
+            </Col>
+          </Row>
+          <Modal
```
Comment (tayyabshabab) [inline src/App.js]: `Modal` can be a separate component

File: src/components/GoogleMap/GoogleMap.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  };
+
+  render() {
+    const { locations } = this.props;
+    return (
+      <div
+        style={{
+          position: "relative",
+          margin: "0",
+          height: "50vh"
```
Comment (tayyabshabab) [inline src/components/GoogleMap/GoogleMap.js]: Use same standards. Either `vh` or `px`

File: src/components/GoogleMap/GoogleMap.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+          style={mapStyles}
+          initialCenter={{
+            lat: 51.198657,
+            lng: 10.606421
+          }}
+          onReady={this.props.handleMapReady}
+        >
+          {locations &&
+            locations.length > 0 &&
+            locations.map(location => (
```
Comment (tayyabshabab) [inline src/components/GoogleMap/GoogleMap.js]: `{locations && locations.map(location => (`
The above code will work the same

File: src/components/Location/LocationsListing.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+import { Card, Col, Row, Button, Popconfirm } from "antd";
+
+import "./LocationsListing.scss";
+
+export class LocationsListing extends Component {
+  render() {
+    const { locations, onEdit } = this.props;
+    const locationsList =
+      locations &&
+      locations.length > 0 &&
```
Comment (tayyabshabab) [inline src/components/Location/LocationsListing.js]: `locations.length` is not needed

File: src/components/Location/SearchAddress.js [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+    this.props.handleError(error);
+    clearSuggestions();
+  };
+
+  componentWillMount = () => {
+    this.setState({ address: this.props.address });
+  };
+
+  render() {
+    const address = this.state.address ? this.state.address : this.props.address;
```
Comment (tayyabshabab) [inline src/components/Location/SearchAddress.js]: Use `const address = this.state.address || this.props.address;`

File: src/components/Location/LocationsListing.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
+import React, { Component } from "react";
+import { Card, Col, Row, Button, Popconfirm } from "antd";
+
+import "./LocationsListing.scss";
+
+export class LocationsListing extends Component {
```
Comment (tayyabshabab) [inline src/components/Location/LocationsListing.js]: Use functional component

File: src/App.js (hunk 4 above)
Comment (tayyabshabab) [reply src/App.js]: Leave it like it is for now as it requires more changes 😆

File: src/components/GoogleMap/GoogleMap.js (hunk 6 above)
Comment (ab-mannan) [reply src/components/GoogleMap/GoogleMap.js]: Done

**Pull request merged**